### ISSUE TITLE
dependabot validator has gone missing

### ISSUE THREAD

**Issue by saramah** [dependabot/dependabot-core#4605]
There used to be a dependabot config validator at `https://dependabot.com/docs/config-file/validator/`, but it has gone missing. Is there a replacement?

references:
https://stackoverflow.com/questions/63943201/how-do-i-test-dependabot-before-merging-config
https://github.community/t/how-to-validate-dependabot-config-file/218636

**Package ecosystem**

**Package manager version**

**Language version**

**Manifest location and content prior to update**

**dependabot.yml content**

**Updated dependency**

**What you expected to see, versus what you actually saw**

**Native package manager behavior**

**Images of the diff or a link to the PR, issue or logs**

**🕹 Bonus points: Smallest manifest that reproduces the issue**

**Comment by mark-stopka** [comment]
Same issue here...

**Comment by jurre** [comment]
The Dependabot app registers a "CheckRun" in GH whenever the config file changes: https://github.com/dependabot/dependabot-core/runs/4327679442

Does that help?

**Comment by hfhbd** [comment]
@jurre Not really, as the `CheckRun` only runs at the `main` branch, not in a PR. This is even visible in your linked example: 
https://github.com/dependabot/dependabot-core/pull/4454/checks, in the checks there is no validator.

See also: https://github.com/dependabot/dependabot-core/issues/3878

**Comment by jurre** [comment]
@hfhbd ah yes, that's still not working :( I'll see about getting that prioritized.

**Comment by jurre** [comment]
In an effort to be transparent, I just investigated this a little and it seems the issue that prevents the check from running on PRs is fairly deep inside GitHub itself, and will require us to rejig some things internally so unfortunately not an easy fix. Will still try to get it prioritized either way, but just might take a little longer for us to roll out a fix because it involves changing a bunch of moving parts :(

**Comment by brrygrdn** [comment]
👋🏻  Just removing the needs-triage as we've been looking at this internally. There is some work we want to do around Dependabot configuration and enablement, so I've made a note of this problem as part of that. I don't have anything to share right now, but this is something we'd like to do better on.

**Comment by jlosito** [comment]
@jurre @brrygrdn I understand that some additional rejigging would need to be done, but would there be the possibility to release some sort of cli tool which people could run in order to validate the configuration file? Something such as `dp-lint /path/to/.github/dependabot.yml`.

**Comment by jeffwidman** [comment]
I realize the original post title was asking for a stand-alone validator, but I suspect the real intent here is to make it possible to validate the config before it lands on `master`/`main`. 

That could be implemented via a standalone validator, but as discussed later in this issue's comments, it could also be via a check that runs on the PR before merge. 

So I reworded the title to focus on the problem, rather than describing only one possible fix implementation.

And yes, we do very much want to resolve the hurdles mentioned in https://github.com/dependabot/dependabot-core/issues/4605#issuecomment-1023250637 and get this resolved.

**Comment by jeffwidman** [comment]
I just noticed that the GH action that validates `dependabot.yaml` just ran on a PR: 
https://github.com/dependabot/dependabot-core/pull/5513/checks?check_run_id=8648478306

I'm a bit surprised because AFAIK we didn't make the the internal changes described in https://github.com/dependabot/dependabot-core/issues/4605#issuecomment-1023250637.

The only thing that is different from a normal PR is that I updated the PR with a rebase off `main`... so perhaps that triggered it? 😕

**Comment by jeffwidman** [comment]
I just noticed the check also ran on this PR branch: https://github.com/dependabot/cli/pull/17

And that PR doesn't touch `dependabot.yml` file... BUT, interestingly, this PR that I merged 15 mins before _does change the dependabot file: https://github.com/dependabot/cli/pull/18

**Comment by jeffwidman** [comment]
As a quick update on this issue, this is still on our radar. Some of the internal changes that are needed to unblock the work on this are happening this quarter. Again, that'd only unblock the actual work, but it's a step in the right direction.

**Comment by maxmoehl** [comment]
I stumbled across this issue because I'm about to add dependabot to a repo and wanted to verify my config. Initially the check didn't show but after I force-pushed the branch once the check suddenly appeared. Don't know if this is reproducible: cloudfoundry/haproxy-boshrelease#367

**Comment by MuhammedMahrous** [comment]
Any updates on this?

**Comment by wesley-harding** [comment]
This is currently hindering my adoption of Dependabot. It's a huge pain to have to go through our entire SDLC just to learn a validation failed.

Further complicating this is the docs are, well, rather underwhelming. Docs give direction indication, but there's a lot left to imagination.

**Comment by larouxn** [comment]
👋 I'm not sure exactly how this GitHub Action is run/added to our merge flow but it does appear possible to validate `dependabot.yml` changes prior to merging to `main`. In the following case I mispelled `package-ecosystem`.
<img width="1081" alt="Screenshot 2023-12-19 at 7 35 41" src="https://github.com/dependabot/dependabot-core/assets/1557529/5d1fd07e-4dfb-435c-b1aa-1a8cc9dd0d87">

Then corrected it.
<img width="370" alt="Screenshot 2023-12-19 at 7 34 01" src="https://github.com/dependabot/dependabot-core/assets/1557529/e53f0a40-8ec8-42fc-b6ba-0c0198c88493">

**Comment by ipowellBT** [comment]
> 👋 I'm not sure exactly how this GitHub Action is run/added to our merge flow but it does appear possible to validate `dependabot.yml` changes prior to merging to `main`. In the following case I mispelled `package-ecosystem`. <img alt="Screenshot 2023-12-19 at 7 35 41" width="1081" src="https://private-user-images.githubusercontent.com/1557529/291587385-5d1fd07e-4dfb-435c-b1aa-1a8cc9dd0d87.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2MDkzODIsIm5iZiI6MTcwNTYwOTA4MiwicGF0aCI6Ii8xNTU3NTI5LzI5MTU4NzM4NS01ZDFmZDA3ZS00ZGZiLTQzNWMtYjFhYS0xYThjYzlkZDBkODcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMThUMjAxODAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzcwNDU3YTI3ZmQwZmFmNDU5OWE0MmIxNTUzY2E5ZGIxOWM0ODFhN2RlYjM4Zjc3ZDMxM2EzZTVjMDcyMzM1OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1jLQURxOuxZyJpb0wL0_nUFXTqiT2IpUj2zfSeGd1ZU">
> 
> Then corrected it. <img alt="Screenshot 2023-12-19 at 7 34 01" width="370" src="https://private-user-images.githubusercontent.com/1557529/291587393-e53f0a40-8ec8-42fc-b6ba-0c0198c88493.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU2MDkzODIsIm5iZiI6MTcwNTYwOTA4MiwicGF0aCI6Ii8xNTU3NTI5LzI5MTU4NzM5My1lNTNmMGE0MC04ZWM4LTQyZmMtYjZiYS0wYzAxOThjODg0OTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDExOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMThUMjAxODAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGMxOWE4YjlmZWM4YzNlZjc2NmUyNzM1NzFkZGUxNDEyM2UxZmRiNDQzZjY1MTI4ZmRkYTUxYzAxNmZkZjFmMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6cYl8CBOHT1uKAlkS2kBG8hRn6q2bQLFtZb2NNzmmXA">

It's important to note that while dependabot supports `dependabot.yml` and `dependabot.yaml`, only dependabot configs named `dependabot.yml` are validated.

**Comment by nickzelei** [comment]
I ran into this myself after merging a bad config into main (which was previously valid somehow).
I put up another PR to try and fix it and suddenly the same job that didn't run on my previous PR is now running to validate the dependabot config.

Some weird behavior here. Why didn't it run previously on the PR that ended up breaking, but now it runs on the subsequent PR to fix it?

**Comment by bugron** [comment]
I'm sharing this here in case anyone finds it useful.


I also needed to validate a Dependabot config file without committing to the default branch first so I've published a CLI tool [@bugron/validate-dependabot-yaml](https://www.npmjs.com/package/@bugron/validate-dependabot-yaml) that does exactly that and more. It is capable of logging validation errors both in Markdown and JSON so can naturally used in GH Actions for sticky comments and post-processing errors. Note, the tool isn't perfect, but should do the job.

**Comment by jpickwell** [comment]
> I'm sharing this here in case anyone finds it useful.
> 
> If your Dependabot config file is located at `.github/dependabot.yml` (notice the extension) simply run the following command in the root of your repository. See the [docs](https://www.npmjs.com/package/@bugron/validate-dependabot-yaml) regarding passing another path.
> 
> ```shell
> npx -y @bugron/validate-dependabot-yaml@latest
> ```
> 
> I also needed to validate a Dependabot config file without committing to the default branch first so I've published a CLI tool [@bugron/validate-dependabot-yaml](https://www.npmjs.com/package/@bugron/validate-dependabot-yaml) that does exactly that and more. It is capable of logging validation errors both in Markdown and JSON so can naturally used in GH Actions for sticky comments and post-processing errors. Note, the tool isn't perfect, but should do the job.

This is cool. However, if you're using VSCode, then you can get the same thing using the [YAML extension by Red Hat](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) along with a comment at the top of your YAML file:

```yaml
# yaml-language-server: $schema=https://json.schemastore.org/dependabot-2.0.json
```

**Comment by jpickwell** [comment]
I would like to have a standalone validator so I can validate my config before pushing it.

**Comment by jrfnl** [comment]
Came here for the same, as I just got caught out by a config key which is supported in some ecosystems, but not in others and now Dependabot is silently failing (<= this is the most annoying thing about no validation in PRs).

Even more annoying, I only noticed this after pulling the same PR to over a dozen repos... so now I have to go back and fix it everywhere, wasting time I could have spend on something useful.

Unfortunately, @bugron's tool does not catch that kind of error (I just tested it), though kudos on creating a validation tool!

Either way, this feature request to have the ability to validate Dependabot config files when they are being changed in a PR via some check on the PR is still very very very valid.

**Comment by yeikel** [comment]
Yep, I saw the same with https://github.com/dependabot/dependabot-core/issues/13121

**Comment by bugron** [comment]
@yeikel published a new version of https://www.npmjs.com/package/@bugron/validate-dependabot-yaml, which addresses #13121.
@jrfnl is your issue the same one reported in #13121?

**Comment by yeikel** [comment]
> @yeikel published a new version of https://www.npmjs.com/package/@bugron/validate-dependabot-yaml, which addresses #13121.
> @jrfnl is your issue the same one reported in #13121?

Thanks, I think that it may help some people

To clarify, I think that #13121 should be handled by the Dependabot job or API as, in my case, it is considered a "valid" job when it's actually invalid

**Comment by jrfnl** [comment]
> @jrfnl is your issue the same one reported in #13121?

@bugron Similar, but not the same. Left a comment in #13121

**Comment by bugron** [comment]
> > [@yeikel](https://github.com/yeikel) published a new version of https://www.npmjs.com/package/@bugron/validate-dependabot-yaml, which addresses [#13121](https://github.com/dependabot/dependabot-core/issues/13121).
> > [@jrfnl](https://github.com/jrfnl) is your issue the same one reported in [#13121](https://github.com/dependabot/dependabot-core/issues/13121)?
> 
> Thanks, I think that it may help some people
> 
> To clarify, I think that [#13121](https://github.com/dependabot/dependabot-core/issues/13121) should be handled by the Dependabot job or API as, in my case, it is considered a "valid" job when it's actually invalid

Yes, I understand and I agree that all these validations should be handled by Dependabot itself. However, by the number of issues and PRs I don't think this is going to be handled soon even if someone volunteers to implement the changes.
So for me it's much easier to update https://www.npmjs.com/package/@bugron/validate-dependabot-yaml, which can be used in CI/CD as well as locally to validate Dependabot configuration files.

**Comment by yeikel** [comment]
Another example where the built-in failed short: https://github.com/dependabot/dependabot-core/issues/13141

**Comment by tvolk131** [comment]
Was this quietly fixed? I just noticed that in a PR where I added a `dependabot.yml` file, a validation was run on it:

<img width="692" height="298" alt="Image" src="https://github.com/user-attachments/assets/24d9630d-b06a-4c18-863e-8b793884c4eb" />

**Comment by sigmavirus24** [comment]
I don't see a similar check. Maybe only for private/paid repos?

**Comment by rocklan** [comment]
This would be great if this feature existed so we don't merge broken dependabot config file updates without noticing.
